app/controllers/owners_controller.rb

```diff
@@ -0,0 +1,59 @@
+class OwnersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_owner, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @owners = Owner.order(created_at: :desc)
+  end
+
+  def new
+    if current_user.owner == nil
+      @owner = current_user.build_owner
+    else
+      redirect_to owner_path(current_user.owner)
+    end
+  end
+
+  def create
+      @owner = current_user.build_owner(owner_params)
+      if @owner.save
+        flash[:notice] = "Owner created"
+        redirect_to @owner
+      else
+        render 'new'
+      end
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    if @owner.update_attributes(owner_params)
```

Comment:
Similarly, you need to define `@owner` here in order to manipulate it in the next line.